### DISCUSSION TITLE
Clarification  on installation docs

### DISCUSSION THREAD

**Discussion by pmkap** [astral-sh/rye#1090] [Q&A] [answered]
From the installations docs:

> Alternatively if you don't trust this approach, you can download the latest release binary. On first run it will install itself.
>
>    [rye-x86_64-linux.gz](https://github.com/astral-sh/rye/releases/latest/download/rye-x86_64-linux.gz) for 64bit Intel computers
>    [rye-aarch64-linux.gz](https://github.com/astral-sh/rye/releases/latest/download/rye-aarch64-linux.gz) for 64bit ARM computers

What does "install itself on first run" mean?

Copy the binary into my path, modify .profile or .bashrc, download python stuff, etc?

Thanks a lot for the clarification.

**Comment by bluss** [comment, marked as answer]
Install on first run means that it will start the installer if it's not already installed into where it expects to be (in $RYE_HOME or ~/.rye). The installer is the command `rye self install`.  It sounds like it to me that you don't need to copy it anywhere, just run it where it was downloaded.

To avoid this, run it will RYE_NO_AUTO_INSTALL=1  set which is documented here https://rye-up.com/guide/installation/#updating-rye
